types/pino/index.d.ts

```diff
@@ -20,7 +20,7 @@
 import stream = require("stream");
 import http = require("http");
 import { EventEmitter } from "events";
-import SonicBoom = require("sonic-boom");
+import SonicBoom from "sonic-boom";
```

Comment:
Wouldn't this break types for people using Pino with older sonic-boom?

Reply:
It very well may.  I've generally thought the types here were optimized for folks running on the latest versions.  Would it make sense to treat this as a semver major for `@types/pino`, setting the expectation it requires a minimum version of `pino` to match?

Reply:
It does look like a semver, yes.
That said, after https://github.com/pinojs/pino/pull/1037 typings for Pino itself should be obsoletted.

Reply:
As far as I understand it, the stance on TypeScript version support is that all versions younger than 2 years are supported and tested. This is exactly what `dts-lint` is testing against. If you don't produce a lint issue with this change then it should be fine.

Reply:
In this case it is related not to TS version per se, but to changes in how `sonic-boom` 2.0.0 exports its types, which is a breaking change.

Reply:
>  Would it make sense to treat this as a semver major for `@types/pino`

You can't make an independent semver major for an @types package. It has to stay in sync with the package it types for the major and minor versions, only the patch level may diverge from `pino`'s version.

Reply:
> That said, due to pinojs/pino#1037, after next semver major is out, typings for Pino itself should be deprecated.

Now that pinojs/pino#1037 is merged. I think it makes the most sense to wait until it's published and then remove both @types/pino and @types/sonic-boom at the same time.

Reply:
Removing `pino` types means that `@types/pino` will have to be allowed in `DefinitelyTyped-tools`, there are still multiple dependencies on `pino` types in the DT tree which will have to explicitly depend on them. All of them will have to be fixed.

Following libs have a direct dependency on `pino` types:
- `hapi-pino`
- `koa-pino-logger`
- `pino-http`
- `pino-multi-stream`
- `pino-pretty`

Following libs depend on `pino` types in their tests, here either the tests have to be adapted or the explicit dependency introduced:
- `ari-client`
- `express-pino-logger`

But all in all this really sounds like the best way froward.

Reply:
I concur. Please review recent removal of `mongodb` types:
 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54510